cmd/juju/controller/showcontroller.go

```diff
@@ -167,19 +169,25 @@ func (c *showControllerCommand) Run(ctx *cmd.Context) error {
 				details.Errors = append(details.Errors, err.Error())
 				continue
 			}
-		}
 
-		var mongoVersion string
-		// Check to see if we can call the mongo version. Older versions of the
-		// api do not support mongo version, so we should even prevent the call.
-		if apiVersion := client.BestAPIVersion(); apiVersion >= 6 {
-			mongoVersion, err = client.MongoVersion()
-			if err != nil {
-				details.Errors = append(details.Errors, err.Error())
-				continue
+			// Check to see if we can call the mongo version. Older versions of the
+			// api do not support mongo version, so we should even prevent the call.
+			if apiVersion := client.BestAPIVersion(); apiVersion >= 6 {
+				mongoVersion, err = client.MongoVersion()
+				if err != nil {
+					details.Errors = append(details.Errors, err.Error())
+					continue
+				}
 			}
 		}
 
+		// Fetch identityURL if the apiserver supports it
+		identityURL, err := client.IdentityProviderURL()
+		if err != nil && !errors.IsNotSupported(err) {
```

Comment:
I do find this much cleaner, maybe we should change the mongo version so it uses the same pattern?

Reply:
Sure, let me update my PR